src/components/Seat/JetsSeat.integration.test.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { MockJetsContextProvider } from '../../__mocks__/MockJetsContext';
+
+import { paramsData, seatDataFirst } from './__fixtures__';
+import { JetsSeat } from './index';
+
+const setup = ({ data = {}, config = {}, params = {}, events = {} } = {}) => ({
+  user: userEvent.setup(),
+  ...render(
+    <MockJetsContextProvider config={config} events={events} params={{ ...paramsData(params) }}>
+      <JetsSeat data={data} />
+    </MockJetsContextProvider>
+  ),
+});
+
+describe('JetsSeat', () => {
+  describe('when any seat type is rendered', () => {
+    it('should trigger onClick handlers when clicked', async () => {
+      const onSeatClick = jest.fn();
+
+      const { user } = setup({
+        data: seatDataFirst(),
+        events: { onSeatClick },
+      });
+
+      await user.click(screen.getByText(/1A/));
+      expect(onSeatClick).toHaveBeenCalledTimes(1);
+      expect(onSeatClick).toHaveBeenCalledWith(seatDataFirst(), expect.any(Object), expect.any(Object));
+    });
+  });
+
+  describe('when hover mode is turned on', () => {
+    it('should trigger onMouseEnter and onMouseLeave handlers', async () => {
+      const showTooltip = jest.fn();
+      const onTooltipClose = jest.fn();
+
+      const { user } = setup({
+        data: seatDataFirst(),
+        events: { onTooltipClose, showTooltip },
+        params: { tooltipOnHover: true },
+      });
+
+      const seatEl = screen.getByText(/1A/);
+
+      await user.hover(seatEl);
+      expect(showTooltip).toHaveBeenCalledTimes(1);
+      expect(showTooltip).toHaveBeenCalledWith(seatDataFirst(), expect.any(Object), expect.any(Object));
+
+      await user.unhover(seatEl);
+      expect(onTooltipClose).toHaveBeenCalledTimes(1);
+      expect(onTooltipClose).toHaveBeenCalledWith(seatDataFirst(), expect.any(Object), expect.any(Object));
+    });
+  });
+
+  describe('when a seat should be jumped to', () => {
```

Comment:
What about 'when a seat label is specified to jump to'

Reply:
I've added the test to the top section as that's the default render/